examples/gno.land/p/demo/ownable/exts/authorizable/authorizable.gno

```diff
@@ -0,0 +1,90 @@
+// Package authorizable is an extension of p/demo/ownable;
+// It allows the user to instantiate an Authorizable struct, which extends
+// p/demo/ownable with a list of users that are authorized for something.
+// By using authorizable, you have a superuser (ownable), as well as another
+// authorization level, which can be used for adding moderators or similar to your realm.
+package authorizable
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ownable"
+	"gno.land/p/demo/ufmt"
+)
+
+type Authorizable struct {
+	*ownable.Ownable           // owner in ownable is superuser
+	authorized       *avl.Tree // std.Addr > struct{}{}
+}
+
+func NewAuthorizable() *Authorizable {
+	a := &Authorizable{
+		ownable.New(),
+		avl.NewTree(),
+	}
+
+	// Add owner to auth list
+	a.authorized.Set(a.Owner().String(), struct{}{})
+	return a
+}
+
+func NewAuthorizableWithAddress(addr std.Address) *Authorizable {
+	a := &Authorizable{
+		ownable.NewWithAddress(addr),
+		avl.NewTree(),
+	}
+
+	// Add owner to auth list
+	a.authorized.Set(a.Owner().String(), struct{}{})
+	return a
+}
+
+func (a *Authorizable) AddToAuthList(addr std.Address) error {
+	if err := a.CallerIsOwner(); err != nil {
+		return ErrNotSuperuser
+	}
+
+	if _, exists := a.authorized.Get(addr.String()); exists {
+		return ErrAlreadyInList
+	}
+
+	a.authorized.Set(addr.String(), struct{}{})
+
+	return nil
+}
+
+func (a *Authorizable) DelFromAuthList(addr std.Address) error {
+	if err := a.CallerIsOwner(); err != nil {
+		return ErrNotSuperuser
+	}
+
+	if !a.authorized.Has(addr.String()) {
+		return ErrNotInAuthList
+	}
+
+	if _, removed := a.authorized.Remove(addr.String()); !removed {
+		str := ufmt.Sprintf("authorizable: could not remove %s from auth list", addr.String())
+		panic(str)
+	}
+
+	return nil
+}
+
+func (a Authorizable) CallerOnAuthList() error {
```

Comment:
The idea was to follow how Ownable checks for the permissions (it also returns an error if the caller is not the owner); not really sure what the right option is; I think both can make sense, but I tip towards `error`.

Reply:
I think it should be up to the caller what to do with this information. If they want to return an error because the caller isn't on the auth list, let them. I'm not sure what the benefit is of returning an error over a boolean, especially when the function is named as if it should return a boolean.

Reply:
Hmmmmm, I'm not sure. I purposefully want to have a package that will be strict about what it does and force people to use it a specific way, especially when it's a critical package like this for managing admin stuff. What do you think a better name would be?

Reply:
That's fine. Then maybe the name can prefixed with something like `check` so it isn't named like it should return a boolean value.